### DISCUSSION TITLE
retrieving capacitance matrix after simulation with Maxwell3d ElectroStatic

### DISCUSSION THREAD

**Discussion by geexie** [ansys/pyaedt#5589] [Q&A] [answered]
I simulate capacitance matrix for my qubit design with Maxwell3d. I export model from GDS file, and assign the excitations and matrix. What is the proper way to retrieve capacitance matrix after simulation?

Currently I use the following code 

```
expressions = m3d.post.available_report_quantities(solution='Setup1 : LastAdaptive', quantities_category='C')
solution = m3d.post.get_solution_data(expressions=expressions, report_category='C')
print ("solution = ", solution)
solution.export_data_to_csv(os.path.join(proj_dir, "capacitance.csv"))
```

which gives me table with individual element of matrix as well as some non relevant to it information, but I believe it's might be better/easier solution to do so.

```
extent_y_pos [mm];extent_x_size [mm];extent_y_size [mm];signal1_thickness [mm];signal1_lower_elevation [mm];extent_x_pos [mm];C1.C(E1,E1) [pF];C1.C(E1,E2) [pF];C1.C(E1,E3) [pF];C1.C(E2,E1) [pF];C1.C(E2,E2) [pF];C1.C(E2,E3) [pF];C1.C(E3,E1) [pF];C1.C(E3,E2) [pF];C1.C(E3,E3) [pF]
1.0;-1.0;-1.0;1.0000000000000001e-07;0.0;1000.0;0.0948967210541557;-0.0693145511740844;-0.0255821698800632;-0.0693145511740844;0.13952355397594401;-0.0702090028019355;-0.0255821698800632;-0.0702090028019355;0.0957911726819914

```

Manually I can export capacitance matrix to a txt file in a format like following

```
DesignVariation : extent_x_pos='1000mm' extent_x_size='-1000mm' extent_y_pos='1000mm' extent_y_size='-1000mm' Signal1_lower_elevation='0mm'
Solution : Setup1 : LastAdaptive
Parameter : Matrix1
Capacitance Unit: pF

Capacitance
		V0	V1	V2
	V0	0.12226	-0.059571	-0.062688
	V1	-0.059571	0.081662	-0.022091
	V2	-0.062688	-0.022091	0.084779

Capacitive Coupling Coefficient
		V0	V1	V2
	V0	1	-0.59619	-0.61574
	V1	-0.59619	1	-0.2655
	V2	-0.61574	-0.2655	1
```

I'm wondering if I can do something similar in pyaedt API.

PS: [full code snippet](https://github.com/geexie/QmQ/blob/main/m3d.py) for the example

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @geexie ,

Once youi get the data: solution = m3d.post.get_solution_data(expressions=expressions, report_category='C')

You do not need to export to data to a file, "solution" is an object with ALL the information, I encourage you to debug and check what it is inside (dir(solution)), for instance you have solution.full_matrix_real_imag , but there are more advanced features.
